pint_pandas/testsuite/test_pandas_extensiontests.py

```diff
@@ -289,7 +293,9 @@ def test_in_numeric_groupby(self, data_for_grouping):
         )
         result = df.groupby("A").sum().columns
 
-        if data_for_grouping.dtype._is_numeric:
+        # FIXME: Why dies C get included for e.g. PandasDtype('complex128') but not for Float64Dtype()? This seems buggy,
+        # but very hard for us to fix...
+        if df.B.isna().sum() == 0 or isinstance(df.B.values.data.dtype, pd.core.dtypes.dtypes.PandasDtype):
             expected = pd.Index(["B", "C"])
```

Comment:
this is odd, I had a start at adding dtypes for other fixtures, but initialising with a list instead of pd.array,

https://github.com/andrewgsavage/pint-pandas/commit/94e764b614f2c79367ace58a7e49533f517b7a8c

this test didn't have any problems

Reply:
This does not happen when you make a raw dataframe (no pint stuff):

```
>>> d = pd.DataFrame({"a": pd.Series([1, np.nan], dtype=pd.Float64Dtype()), "b": pd.Series([1, np.nan], dtype=np.complex128)})
>>> d.sum()
a    1.0+0.0j
b    1.0+0.0j
dtype: complex128
>>> d["x"] = 1
>>> d.groupby("x").sum()
     a   b
x         
1  1.0 NaN

```

Reply:
So what exactly does a series backed by a PintArray do when you call sum() on it? Does it use the _reduce() method of the PintArray?